internal/ibmcloud-janitor/resources/vpc_lbs.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"strings"
+	"time"
+
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type VPCLoadBalancer struct{}
+
+var (
+	lbTimeout      = time.Minute * 10
+	lbWaitTime     = time.Minute * 1
+	resourceLogger *logrus.Entry
+)
+
+// Cleans up the load balancers in a given region
+func (VPCLoadBalancer) cleanup(options *CleanupOptions) error {
+	resourceLogger = logrus.WithFields(logrus.Fields{"resource": options.Resource.Name})
+	resourceLogger.Info("Cleaning up the load balancers")
+	client, err := NewVPCClient(options)
+	if err != nil {
+		return errors.Wrap(err, "couldn't create VPC client")
+	}
+
+	var deletedLBList = make(map[string]bool)
+
+	f := func(start string) (bool, string, error) {
+		listLbOpts := &vpcv1.ListLoadBalancersOptions{}
+		if start != "" {
+			listLbOpts.Start = &start
+		}
+
+		loadBalancers, _, err := client.ListLoadBalancers(listLbOpts)
+		if err != nil {
+			return false, "", errors.Wrap(err, "failed to list the load balancers")
+		}
+
+		if loadBalancers == nil || len(loadBalancers.LoadBalancers) <= 0 {
+			resourceLogger.Info("there are no available load balancers to delete")
+			return true, "", nil
+		}
+
+		for _, lb := range loadBalancers.LoadBalancers {
+			if *lb.ResourceGroup.ID == client.ResourceGroupID {
+				if _, err := client.DeleteLoadBalancer(&vpcv1.DeleteLoadBalancerOptions{
+					ID: lb.ID,
+				}); err != nil {
+					return false, "", errors.Wrapf(err, "failed to delete the load balancer %q", *lb.Name)
+				}
+				deletedLBList[*lb.ID] = false
+			}
+			resourceLogger.WithField("name", *lb.Name).Info("load balancer deletetion triggered")
```

Comment:
This line should also be inside if or else it would print for all the LB even if deletion not attempted.

Reply:
yes, missed it. Thanks for pointing it out.